### PR TITLE
Adding new Workflows

### PR DESCRIPTION
**Pull request by puneeth072003**
### Pending Workflows to build
- [x] CPlus
- [ ] Rust
- [x] Golang

### Currently working on Rust workflow

### CODE REVIEW

**Comment by puneeth072003** [comment]
@commjoen C and CPlus both workflows are over right?

**Comment by commjoen** [comment]
Yes

**Comment by puneeth072003** [comment]
> @commjoen Hey, Jeroen I'm getting the same error as I encountered before.
> 
> ```
> Compiling Golang for Linux-x64
> Error: golang/main.go:26:[8](https://github.com/puneeth072003/wrongsecrets-binaries/actions/runs/5256982988/jobs/9499094352#step:4:9): no required module provides package github.com/OWASP/wrongsecrets-binaries/golang/cmd: go.mod file not found in current directory or any parent directory; see 'go help modules'
> Error: Process completed with exit code 1.
> ```
> 
> ### In short: go.mod file not found error
> I did use `cd golang` and tested out the workflows but it is no working
> 
> #### Can you please tell me where I am going wrong?

### Fixed it🥳

File: .github/workflows/compile_golang.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  build-darwin:
+    name: MacOS(Intel&ARM)
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+      - name: compile Golang binary for macOS
```
Comment (commjoen) [inline .github/workflows/compile_golang.yml:25]: Please check https://github.com/OWASP/wrongsecrets-binaries/blob/main/quickbuild.sh#L52: there is no need for docker containers

**Review by commjoen — APPROVED** [review]
Will merge this PR now though Rust is not completed yet. Can you provide a separate PR for that please?

**Pull request merged**